[BUG] org.opensearch.repositories.s3.S3BlobStoreRepositoryTests.testRequestStats is flaky

**Describe the bug**
https://build.ci.opensearch.org/job/gradle-check/28391/ 

So many runs in gradle are failing due to same.

Failure log: 
[2024-05-21T13:54:24,055][INFO ][o.o.t.OpenSearchTestClusterRule] [testRequestStats] [S3BlobStoreRepositoryTests#testRequestStats]: setting up test
[2024-05-21T13:54:24,056][INFO ][o.o.t.InternalTestCluster] [testRequestStats] Setup InternalTestCluster [TEST-TEST_WORKER_VM=[--not-gradle--]-CLUSTER_SEED=[1285764699169022751]-HASH=[D539AA680FC]-cluster] with seed [11D7F42E7EB4E31F] using [0] dedicated cluster-managers, [2] (data) nodes and [1] coord only nodes (min_cluster_manager_nodes are [auto-managed])
[2024-05-21T13:54:24,063][INFO ][o.o.n.Node               ] [testRequestStats] version[3.0.0], pid[36506], build[unknown/unknown/unknown], OS[Mac OS X/14.5/x86_64], JVM[Eclipse Adoptium/OpenJDK 64-Bit Server VM/11.0.23/11.0.23+9]
[2024-05-21T13:54:24,063][INFO ][o.o.n.Node               ] [testRequestStats] JVM home [/Library/Java/JavaVirtualMachines/temurin-11.jdk/Contents/Home]
[2024-05-21T13:54:24,063][DEPRECATION][o.o.d.n.Node             ] [testRequestStats] no-jdk distributions that do not bundle a JDK are deprecated and will be removed in a future release
[2024-05-21T13:54:24,063][INFO ][o.o.n.Node               ] [testRequestStats] JVM arguments [-ea, -Djava.locale.providers=SPI,COMPAT, -Didea.test.cyclic.buffer.size=1048576, -javaagent:/Applications/IntelliJ IDEA CE.app/Contents/lib/idea_rt.jar=54648:/Applications/IntelliJ IDEA CE.app/Contents/bin, -Dfile.encoding=UTF-8]
[2024-05-21T13:54:24,065][INFO ][o.o.p.PluginsService     ] [testRequestStats] no modules loaded
[2024-05-21T13:54:24,065][INFO ][o.o.p.PluginsService     ] [testRequestStats] loaded plugin [org.opensearch.repositories.s3.S3BlobStoreRepositoryTests$TestS3RepositoryPlugin]
[2024-05-21T13:54:24,065][INFO ][o.o.p.PluginsService     ] [testRequestStats] loaded plugin [org.opensearch.script.MockScriptService$TestPlugin]
[2024-05-21T13:54:24,065][INFO ][o.o.p.PluginsService     ] [testRequestStats] loaded plugin [org.opensearch.test.MockHttpTransport$TestPlugin]
[2024-05-21T13:54:24,065][INFO ][o.o.p.PluginsService     ] [testRequestStats] loaded plugin [org.opensearch.test.OpenSearchIntegTestCase$AssertActionNamePlugin]
[2024-05-21T13:54:24,065][INFO ][o.o.p.PluginsService     ] [testRequestStats] loaded plugin [org.opensearch.test.OpenSearchIntegTestCase$TestSeedPlugin]
[2024-05-21T13:54:24,065][INFO ][o.o.p.PluginsService     ] [testRequestStats] loaded plugin [org.opensearch.test.TestGeoShapeFieldMapperPlugin]
[2024-05-21T13:54:24,065][INFO ][o.o.p.PluginsService     ] [testRequestStats] loaded plugin [org.opensearch.test.telemetry.MockTelemetryPlugin]
[2024-05-21T13:54:24,065][INFO ][o.o.p.PluginsService     ] [testRequestStats] loaded plugin [org.opensearch.test.transport.MockTransportService$TestPlugin]
[2024-05-21T13:54:24,065][INFO ][o.o.p.PluginsService     ] [testRequestStats] loaded plugin [org.opensearch.transport.nio.MockNioTransportPlugin]
[2024-05-21T13:54:24,066][INFO ][o.o.e.ExtensionsManager  ] [testRequestStats] ExtensionsManager initialized
[2024-05-21T13:54:24,068][INFO ][o.o.e.NodeEnvironment    ] [testRequestStats] using [1] data paths, mounts [[/System/Volumes/Data (/dev/disk3s5)]], net usable_space [827.4gb], net total_space [926.3gb], types [apfs]
[2024-05-21T13:54:24,069][INFO ][o.o.e.NodeEnvironment    ] [testRequestStats] heap size [9gb], compressed ordinary object pointers [true]
[2024-05-21T13:54:24,071][INFO ][o.o.n.Node               ] [testRequestStats] node name [node_t0], node ID [tGPNeinSQ5WZPUeJ68LKpw], cluster name [TEST-TEST_WORKER_VM=[--not-gradle--]-CLUSTER_SEED=[1285764699169022751]-HASH=[D539AA680FC]-cluster], roles [ingest, remote_cluster_client, data, cluster_manager]
[2024-05-21T13:54:24,098][INFO ][o.o.d.DiscoveryModule    ] [testRequestStats] using discovery type [zen] and seed hosts providers [settings, file]
[2024-05-21T13:54:24,118][WARN ][o.o.g.DanglingIndicesState] [testRequestStats] gateway.auto_import_dangling_indices is disabled, dangling indices will not be automatically detected or imported and must be managed manually
[2024-05-21T13:54:24,127][INFO ][o.o.n.Node               ] [testRequestStats] initialized
[2024-05-21T13:54:24,129][INFO ][o.o.n.Node               ] [testRequestStats] version[3.0.0], pid[36506], build[unknown/unknown/unknown], OS[Mac OS X/14.5/x86_64], JVM[Eclipse Adoptium/OpenJDK 64-Bit Server VM/11.0.23/11.0.23+9]
[2024-05-21T13:54:24,129][INFO ][o.o.n.Node               ] [testRequestStats] JVM home [/Library/Java/JavaVirtualMachines/temurin-11.jdk/Contents/Home]
[2024-05-21T13:54:24,129][INFO ][o.o.n.Node               ] [testRequestStats] JVM arguments [-ea, -Djava.locale.providers=SPI,COMPAT, -Didea.test.cyclic.buffer.size=1048576, -javaagent:/Applications/IntelliJ IDEA CE.app/Contents/lib/idea_rt.jar=54648:/Applications/IntelliJ IDEA CE.app/Contents/bin, -Dfile.encoding=UTF-8]
[2024-05-21T13:54:24,131][INFO ][o.o.p.PluginsService     ] [testRequestStats] no modules loaded
[2024-05-21T13:54:24,131][INFO ][o.o.p.PluginsService     ] [testRequestStats] loaded plugin [org.opensearch.repositories.s3.S3BlobStoreRepositoryTests$TestS3RepositoryPlugin]
[2024-05-21T13:54:24,131][INFO ][o.o.p.PluginsService     ] [testRequestStats] loaded plugin [org.opensearch.script.MockScriptService$TestPlugin]
[2024-05-21T13:54:24,131][INFO ][o.o.p.PluginsService     ] [testRequestStats] loaded plugin [org.opensearch.test.MockHttpTransport$TestPlugin]
[2024-05-21T13:54:24,131][INFO ][o.o.p.PluginsService     ] [testRequestStats] loaded plugin [org.opensearch.test.OpenSearchIntegTestCase$AssertActionNamePlugin]
[2024-05-21T13:54:24,131][INFO ][o.o.p.PluginsService     ] [testRequestStats] loaded plugin [org.opensearch.test.OpenSearchIntegTestCase$TestSeedPlugin]
[2024-05-21T13:54:24,131][INFO ][o.o.p.PluginsService     ] [testRequestStats] loaded plugin [org.opensearch.test.TestGeoShapeFieldMapperPlugin]
[2024-05-21T13:54:24,131][INFO ][o.o.p.PluginsService     ] [testRequestStats] loaded plugin [org.opensearch.test.telemetry.MockTelemetryPlugin]
[2024-05-21T13:54:24,131][INFO ][o.o.p.PluginsService     ] [testRequestStats] loaded plugin [org.opensearch.test.transport.MockTransportService$TestPlugin]
[2024-05-21T13:54:24,131][INFO ][o.o.p.PluginsService     ] [testRequestStats] loaded plugin [org.opensearch.transport.nio.MockNioTransportPlugin]
[2024-05-21T13:54:24,131][INFO ][o.o.e.ExtensionsManager  ] [testRequestStats] ExtensionsManager initialized
[2024-05-21T13:54:24,132][INFO ][o.o.e.NodeEnvironment    ] [testRequestStats] using [1] data paths, mounts [[/System/Volumes/Data (/dev/disk3s5)]], net usable_space [827.4gb], net total_space [926.3gb], types [apfs]
[2024-05-21T13:54:24,133][INFO ][o.o.e.NodeEnvironment    ] [testRequestStats] heap size [9gb], compressed ordinary object pointers [true]
[2024-05-21T13:54:24,134][INFO ][o.o.n.Node               ] [testRequestStats] node name [node_t1], node ID [zEe9xRGYRoiU7kpgA-aeNQ], cluster name [TEST-TEST_WORKER_VM=[--not-gradle--]-CLUSTER_SEED=[1285764699169022751]-HASH=[D539AA680FC]-cluster], roles [ingest, remote_cluster_client, data, cluster_manager]
[2024-05-21T13:54:24,149][INFO ][o.o.d.DiscoveryModule    ] [testRequestStats] using discovery type [zen] and seed hosts providers [settings, file]
[2024-05-21T13:54:24,155][WARN ][o.o.g.DanglingIndicesState] [testRequestStats] gateway.auto_import_dangling_indices is disabled, dangling indices will not be automatically detected or imported and must be managed manually
[2024-05-21T13:54:24,160][INFO ][o.o.n.Node               ] [testRequestStats] initialized
[2024-05-21T13:54:24,162][INFO ][o.o.n.Node               ] [testRequestStats] version[3.0.0], pid[36506], build[unknown/unknown/unknown], OS[Mac OS X/14.5/x86_64], JVM[Eclipse Adoptium/OpenJDK 64-Bit Server VM/11.0.23/11.0.23+9]
[2024-05-21T13:54:24,162][INFO ][o.o.n.Node               ] [testRequestStats] JVM home [/Library/Java/JavaVirtualMachines/temurin-11.jdk/Contents/Home]
[2024-05-21T13:54:24,163][INFO ][o.o.n.Node               ] [testRequestStats] JVM arguments [-ea, -Djava.locale.providers=SPI,COMPAT, -Didea.test.cyclic.buffer.size=1048576, -javaagent:/Applications/IntelliJ IDEA CE.app/Contents/lib/idea_rt.jar=54648:/Applications/IntelliJ IDEA CE.app/Contents/bin, -Dfile.encoding=UTF-8]
[2024-05-21T13:54:24,163][INFO ][o.o.p.PluginsService     ] [testRequestStats] no modules loaded
[2024-05-21T13:54:24,163][INFO ][o.o.p.PluginsService     ] [testRequestStats] loaded plugin [org.opensearch.repositories.s3.S3BlobStoreRepositoryTests$TestS3RepositoryPlugin]
[2024-05-21T13:54:24,163][INFO ][o.o.p.PluginsService     ] [testRequestStats] loaded plugin [org.opensearch.script.MockScriptService$TestPlugin]
[2024-05-21T13:54:24,163][INFO ][o.o.p.PluginsService     ] [testRequestStats] loaded plugin [org.opensearch.test.MockHttpTransport$TestPlugin]
[2024-05-21T13:54:24,163][INFO ][o.o.p.PluginsService     ] [testRequestStats] loaded plugin [org.opensearch.test.OpenSearchIntegTestCase$AssertActionNamePlugin]
[2024-05-21T13:54:24,163][INFO ][o.o.p.PluginsService     ] [testRequestStats] loaded plugin [org.opensearch.test.OpenSearchIntegTestCase$TestSeedPlugin]
[2024-05-21T13:54:24,163][INFO ][o.o.p.PluginsService     ] [testRequestStats] loaded plugin [org.opensearch.test.TestGeoShapeFieldMapperPlugin]
[2024-05-21T13:54:24,163][INFO ][o.o.p.PluginsService     ] [testRequestStats] loaded plugin [org.opensearch.test.telemetry.MockTelemetryPlugin]
[2024-05-21T13:54:24,163][INFO ][o.o.p.PluginsService     ] [testRequestStats] loaded plugin [org.opensearch.test.transport.MockTransportService$TestPlugin]
[2024-05-21T13:54:24,163][INFO ][o.o.p.PluginsService     ] [testRequestStats] loaded plugin [org.opensearch.transport.nio.MockNioTransportPlugin]
[2024-05-21T13:54:24,163][INFO ][o.o.e.ExtensionsManager  ] [testRequestStats] ExtensionsManager initialized
[2024-05-21T13:54:24,165][INFO ][o.o.e.NodeEnvironment    ] [testRequestStats] using [1] data paths, mounts [[/System/Volumes/Data (/dev/disk3s5)]], net usable_space [827.4gb], net total_space [926.3gb], types [apfs]
[2024-05-21T13:54:24,165][INFO ][o.o.e.NodeEnvironment    ] [testRequestStats] heap size [9gb], compressed ordinary object pointers [true]
[2024-05-21T13:54:24,166][INFO ][o.o.n.Node               ] [testRequestStats] node name [node_t2], node ID [b_-9xXmqQTy5VW6fvCGllw], cluster name [TEST-TEST_WORKER_VM=[--not-gradle--]-CLUSTER_SEED=[1285764699169022751]-HASH=[D539AA680FC]-cluster], roles []
[2024-05-21T13:54:24,178][INFO ][o.o.d.DiscoveryModule    ] [testRequestStats] using discovery type [zen] and seed hosts providers [settings, file]
[2024-05-21T13:54:24,183][WARN ][o.o.g.DanglingIndicesState] [testRequestStats] gateway.auto_import_dangling_indices is disabled, dangling indices will not be automatically detected or imported and must be managed manually
[2024-05-21T13:54:24,196][INFO ][o.o.n.Node               ] [testRequestStats] initialized
[2024-05-21T13:54:24,200][INFO ][o.o.n.Node               ] [[test_TEST-TEST_WORKER_VM=[--not-gradle--]-CLUSTER_SEED=[1285764699169022751]-HASH=[D539AA680FC]-cluster[T#1]]] starting ...
[2024-05-21T13:54:24,200][INFO ][o.o.n.Node               ] [[test_TEST-TEST_WORKER_VM=[--not-gradle--]-CLUSTER_SEED=[1285764699169022751]-HASH=[D539AA680FC]-cluster[T#2]]] starting ...
[2024-05-21T13:54:24,201][INFO ][o.o.n.Node               ] [[test_TEST-TEST_WORKER_VM=[--not-gradle--]-CLUSTER_SEED=[1285764699169022751]-HASH=[D539AA680FC]-cluster[T#3]]] starting ...
[2024-05-21T13:54:24,230][INFO ][o.o.t.TransportService   ] [[test_TEST-TEST_WORKER_VM=[--not-gradle--]-CLUSTER_SEED=[1285764699169022751]-HASH=[D539AA680FC]-cluster[T#2]]] publish_address {127.0.0.1:56373}, bound_addresses {[::1]:56371}, {127.0.0.1:56373}
[2024-05-21T13:54:24,230][INFO ][o.o.t.TransportService   ] [[test_TEST-TEST_WORKER_VM=[--not-gradle--]-CLUSTER_SEED=[1285764699169022751]-HASH=[D539AA680FC]-cluster[T#3]]] publish_address {127.0.0.1:56375}, bound_addresses {[::1]:56370}, {127.0.0.1:56375}
[2024-05-21T13:54:24,230][INFO ][o.o.t.TransportService   ] [[test_TEST-TEST_WORKER_VM=[--not-gradle--]-CLUSTER_SEED=[1285764699169022751]-HASH=[D539AA680FC]-cluster[T#1]]] publish_address {127.0.0.1:56374}, bound_addresses {[::1]:56372}, {127.0.0.1:56374}
[2024-05-21T13:54:24,232][INFO ][o.o.t.TransportService   ] [[test_TEST-TEST_WORKER_VM=[--not-gradle--]-CLUSTER_SEED=[1285764699169022751]-HASH=[D539AA680FC]-cluster[T#2]]] Remote clusters initialized successfully.
[2024-05-21T13:54:24,232][INFO ][o.o.t.TransportService   ] [[test_TEST-TEST_WORKER_VM=[--not-gradle--]-CLUSTER_SEED=[1285764699169022751]-HASH=[D539AA680FC]-cluster[T#1]]] Remote clusters initialized successfully.
[2024-05-21T13:54:24,331][INFO ][o.o.c.c.Coordinator      ] [node_t0] setting initial configuration to VotingConfiguration{zEe9xRGYRoiU7kpgA-aeNQ,tGPNeinSQ5WZPUeJ68LKpw}
[2024-05-21T13:54:24,469][INFO ][o.o.c.s.MasterService    ] [node_t0] elected-as-cluster-manager ([2] nodes joined)[{node_t1}{zEe9xRGYRoiU7kpgA-aeNQ}{3kVkAprNRqSyxi0UH30Oag}{127.0.0.1}{127.0.0.1:56373}{dimr}{shard_indexing_pressure_enabled=true} elect leader, {node_t0}{tGPNeinSQ5WZPUeJ68LKpw}{dKK-5nlsQmyAqaC35b_Etg}{127.0.0.1}{127.0.0.1:56374}{dimr}{shard_indexing_pressure_enabled=true} elect leader, _BECOME_CLUSTER_MANAGER_TASK_, _FINISH_ELECTION_], term: 1, version: 1, delta: cluster-manager node changed {previous [], current [{node_t0}{tGPNeinSQ5WZPUeJ68LKpw}{dKK-5nlsQmyAqaC35b_Etg}{127.0.0.1}{127.0.0.1:56374}{dimr}{shard_indexing_pressure_enabled=true}]}, added {{node_t1}{zEe9xRGYRoiU7kpgA-aeNQ}{3kVkAprNRqSyxi0UH30Oag}{127.0.0.1}{127.0.0.1:56373}{dimr}{shard_indexing_pressure_enabled=true}}
[2024-05-21T13:54:24,521][INFO ][o.o.c.c.CoordinationState] [node_t1] cluster UUID set to [qqNLMeCoQoODwBGTWlQYsg]
[2024-05-21T13:54:24,521][INFO ][o.o.c.c.CoordinationState] [node_t0] cluster UUID set to [qqNLMeCoQoODwBGTWlQYsg]
[2024-05-21T13:54:24,549][INFO ][o.o.c.s.ClusterApplierService] [node_t1] cluster-manager node changed {previous [], current [{node_t0}{tGPNeinSQ5WZPUeJ68LKpw}{dKK-5nlsQmyAqaC35b_Etg}{127.0.0.1}{127.0.0.1:56374}{dimr}{shard_indexing_pressure_enabled=true}]}, added {{node_t0}{tGPNeinSQ5WZPUeJ68LKpw}{dKK-5nlsQmyAqaC35b_Etg}{127.0.0.1}{127.0.0.1:56374}{dimr}{shard_indexing_pressure_enabled=true}}, term: 1, version: 1, reason: ApplyCommitRequest{term=1, version=1, sourceNode={node_t0}{tGPNeinSQ5WZPUeJ68LKpw}{dKK-5nlsQmyAqaC35b_Etg}{127.0.0.1}{127.0.0.1:56374}{dimr}{shard_indexing_pressure_enabled=true}}
[2024-05-21T13:54:24,556][INFO ][o.o.n.Node               ] [[test_TEST-TEST_WORKER_VM=[--not-gradle--]-CLUSTER_SEED=[1285764699169022751]-HASH=[D539AA680FC]-cluster[T#2]]] started
[2024-05-21T13:54:24,557][INFO ][o.o.c.s.ClusterApplierService] [node_t0] cluster-manager node changed {previous [], current [{node_t0}{tGPNeinSQ5WZPUeJ68LKpw}{dKK-5nlsQmyAqaC35b_Etg}{127.0.0.1}{127.0.0.1:56374}{dimr}{shard_indexing_pressure_enabled=true}]}, added {{node_t1}{zEe9xRGYRoiU7kpgA-aeNQ}{3kVkAprNRqSyxi0UH30Oag}{127.0.0.1}{127.0.0.1:56373}{dimr}{shard_indexing_pressure_enabled=true}}, term: 1, version: 1, reason: Publication{term=1, version=1}
[2024-05-21T13:54:24,564][INFO ][o.o.n.Node               ] [[test_TEST-TEST_WORKER_VM=[--not-gradle--]-CLUSTER_SEED=[1285764699169022751]-HASH=[D539AA680FC]-cluster[T#1]]] started
[2024-05-21T13:54:24,567][INFO ][o.o.c.r.a.DiskThresholdMonitor] [node_t0] skipping monitor as a check is already in progress
[2024-05-21T13:54:24,573][INFO ][o.o.d.PeerFinder         ] [node_t0] setting findPeersInterval to [1s] as node commission status = [true] for local node [{node_t0}{tGPNeinSQ5WZPUeJ68LKpw}{dKK-5nlsQmyAqaC35b_Etg}{127.0.0.1}{127.0.0.1:56374}{dimr}{shard_indexing_pressure_enabled=true}]
[2024-05-21T13:54:24,573][INFO ][o.o.d.PeerFinder         ] [node_t1] setting findPeersInterval to [1s] as node commission status = [true] for local node [{node_t1}{zEe9xRGYRoiU7kpgA-aeNQ}{3kVkAprNRqSyxi0UH30Oag}{127.0.0.1}{127.0.0.1:56373}{dimr}{shard_indexing_pressure_enabled=true}]
[2024-05-21T13:54:24,612][WARN ][o.o.c.r.a.AllocationService] [node_t0] Falling back to single shard assignment since batch mode disable or multiple custom allocators set
[2024-05-21T13:54:24,614][WARN ][o.o.c.r.a.AllocationService] [node_t0] Falling back to single shard assignment since batch mode disable or multiple custom allocators set
[2024-05-21T13:54:24,622][INFO ][o.o.g.GatewayService     ] [node_t0] recovered [0] indices into cluster_state
[2024-05-21T13:54:25,341][INFO ][o.o.c.s.MasterService    ] [node_t0] node-join[{node_t2}{b_-9xXmqQTy5VW6fvCGllw}{GXCiImH3SHmPMtRwT8SmiQ}{127.0.0.1}{127.0.0.1:56375}{shard_indexing_pressure_enabled=true} join existing leader], term: 1, version: 4, delta: added {{node_t2}{b_-9xXmqQTy5VW6fvCGllw}{GXCiImH3SHmPMtRwT8SmiQ}{127.0.0.1}{127.0.0.1:56375}{shard_indexing_pressure_enabled=true}}
[2024-05-21T13:54:25,349][INFO ][o.o.c.s.ClusterApplierService] [node_t2] cluster-manager node changed {previous [], current [{node_t0}{tGPNeinSQ5WZPUeJ68LKpw}{dKK-5nlsQmyAqaC35b_Etg}{127.0.0.1}{127.0.0.1:56374}{dimr}{shard_indexing_pressure_enabled=true}]}, added {{node_t1}{zEe9xRGYRoiU7kpgA-aeNQ}{3kVkAprNRqSyxi0UH30Oag}{127.0.0.1}{127.0.0.1:56373}{dimr}{shard_indexing_pressure_enabled=true},{node_t0}{tGPNeinSQ5WZPUeJ68LKpw}{dKK-5nlsQmyAqaC35b_Etg}{127.0.0.1}{127.0.0.1:56374}{dimr}{shard_indexing_pressure_enabled=true}}, term: 1, version: 4, reason: ApplyCommitRequest{term=1, version=4, sourceNode={node_t0}{tGPNeinSQ5WZPUeJ68LKpw}{dKK-5nlsQmyAqaC35b_Etg}{127.0.0.1}{127.0.0.1:56374}{dimr}{shard_indexing_pressure_enabled=true}}
[2024-05-21T13:54:25,352][INFO ][o.o.n.Node               ] [[test_TEST-TEST_WORKER_VM=[--not-gradle--]-CLUSTER_SEED=[1285764699169022751]-HASH=[D539AA680FC]-cluster[T#3]]] started
[2024-05-21T13:54:25,353][INFO ][o.o.c.s.ClusterApplierService] [node_t1] added {{node_t2}{b_-9xXmqQTy5VW6fvCGllw}{GXCiImH3SHmPMtRwT8SmiQ}{127.0.0.1}{127.0.0.1:56375}{shard_indexing_pressure_enabled=true}}, term: 1, version: 4, reason: ApplyCommitRequest{term=1, version=4, sourceNode={node_t0}{tGPNeinSQ5WZPUeJ68LKpw}{dKK-5nlsQmyAqaC35b_Etg}{127.0.0.1}{127.0.0.1:56374}{dimr}{shard_indexing_pressure_enabled=true}}
[2024-05-21T13:54:25,361][INFO ][o.o.c.s.ClusterApplierService] [node_t0] added {{node_t2}{b_-9xXmqQTy5VW6fvCGllw}{GXCiImH3SHmPMtRwT8SmiQ}{127.0.0.1}{127.0.0.1:56375}{shard_indexing_pressure_enabled=true}}, term: 1, version: 4, reason: Publication{term=1, version=4}
[2024-05-21T13:54:25,362][WARN ][o.o.c.r.a.AllocationService] [node_t0] Falling back to single shard assignment since batch mode disable or multiple custom allocators set
[2024-05-21T13:54:25,362][INFO ][o.o.d.PeerFinder         ] [node_t2] setting findPeersInterval to [1s] as node commission status = [true] for local node [{node_t2}{b_-9xXmqQTy5VW6fvCGllw}{GXCiImH3SHmPMtRwT8SmiQ}{127.0.0.1}{127.0.0.1:56375}{shard_indexing_pressure_enabled=true}]
[2024-05-21T13:54:25,366][WARN ][o.o.c.r.a.AllocationService] [node_t0] Falling back to single shard assignment since batch mode disable or multiple custom allocators set
[2024-05-21T13:54:25,384][DEPRECATION][o.o.d.c.s.Settings       ] [node_t0] [index.force_memory_term_dictionary] setting was deprecated in OpenSearch and will be removed in a future release! See the breaking changes documentation for the next major version.
[2024-05-21T13:54:25,393][INFO ][o.o.p.PluginsService     ] [node_t0] PluginService:onIndexModule index:[P_cgQTOQQZOF3R4VkB2KKQ/e8A3EEySSd6F7PLGHWqSYQ]
[2024-05-21T13:54:25,402][INFO ][o.o.c.m.MetadataIndexTemplateService] [node_t0] adding template [random_index_template] for index patterns [*]
[2024-05-21T13:54:25,429][INFO ][o.o.t.OpenSearchTestClusterRule] [testRequestStats] [S3BlobStoreRepositoryTests#testRequestStats]: all set up test
[2024-05-21T13:54:25,430][INFO ][o.o.r.s.S3BlobStoreRepositoryTests] [testRequestStats] before test
[2024-05-21T13:54:25,432][DEPRECATION][o.o.d.c.s.Settings       ] [testRequestStats] [cache_repository_data] setting was deprecated in OpenSearch and will be removed in a future release! See the breaking changes documentation for the next major version.
[2024-05-21T13:54:25,433][INFO ][o.o.r.RepositoriesService] [node_t0] put repository [cparpfrypg]
[2024-05-21T13:54:25,455][INFO ][o.o.p.PluginsService     ] [node_t0] PluginService:onIndexModule index:[index-no-merges/K5d7xHQuRqa_nrxXAmR0yA]
[2024-05-21T13:54:25,456][INFO ][o.o.c.m.MetadataCreateIndexService] [node_t0] [index-no-merges] creating index, cause [api], templates [random_index_template], shards [1]/[0]
[2024-05-21T13:54:25,457][WARN ][o.o.c.r.a.AllocationService] [node_t0] Falling back to single shard assignment since batch mode disable or multiple custom allocators set
[2024-05-21T13:54:25,482][INFO ][o.o.p.PluginsService     ] [node_t1] PluginService:onIndexModule index:[index-no-merges/K5d7xHQuRqa_nrxXAmR0yA]
[2024-05-21T13:54:25,551][INFO ][o.o.c.r.a.AllocationService] [node_t0] Cluster health status changed from [YELLOW] to [GREEN] (reason: [shards started [[index-no-merges][0]]]).
[2024-05-21T13:54:25,595][WARN ][o.o.c.r.a.AllocationService] [node_t0] Falling back to single shard assignment since batch mode disable or multiple custom allocators set
[2024-05-21T13:54:25,595][INFO ][o.o.t.BackgroundIndexer  ] [testRequestStats] --> creating 2 indexing threads (auto start: [true], numOfDocs: [12757])
[2024-05-21T13:54:25,601][INFO ][o.o.t.BackgroundIndexer  ] [[Thread-332]] **** starting indexing thread 0
[2024-05-21T13:54:25,601][INFO ][o.o.t.BackgroundIndexer  ] [[Thread-333]] **** starting indexing thread 1
[2024-05-21T13:54:25,693][INFO ][o.o.p.PluginsService     ] [node_t0] PluginService:onIndexModule index:[index-no-merges/K5d7xHQuRqa_nrxXAmR0yA]
[2024-05-21T13:54:25,695][INFO ][o.o.c.m.MetadataMappingService] [node_t0] [index-no-merges/K5d7xHQuRqa_nrxXAmR0yA] create_mapping
[2024-05-21T13:54:29,819][INFO ][o.o.t.BackgroundIndexer  ] [[Thread-332]] **** done indexing thread 0  stop: true numDocsIndexed: 12757
[2024-05-21T13:54:29,985][INFO ][o.o.t.BackgroundIndexer  ] [[Thread-333]] **** done indexing thread 1  stop: true numDocsIndexed: 12757
[2024-05-21T13:54:30,691][INFO ][o.o.s.SnapshotsService   ] [node_t0] snapshot [cparpfrypg:snapshot/AHbje9zlR5iEhZJx0K0DXg] started
[2024-05-21T13:54:36,613][INFO ][o.o.s.SnapshotsService   ] [node_t0] snapshot [cparpfrypg:snapshot/AHbje9zlR5iEhZJx0K0DXg] completed with state [SUCCESS]
[2024-05-21T13:54:36,615][INFO ][o.o.c.m.MetadataDeleteIndexService] [node_t0] [index-no-merges/K5d7xHQuRqa_nrxXAmR0yA] deleting index
[2024-05-21T13:54:36,616][WARN ][o.o.c.r.a.AllocationService] [node_t0] Falling back to single shard assignment since batch mode disable or multiple custom allocators set
[2024-05-21T13:54:37,152][WARN ][o.o.c.r.a.AllocationService] [node_t0] Falling back to single shard assignment since batch mode disable or multiple custom allocators set
[2024-05-21T13:54:37,536][WARN ][o.o.c.r.a.AllocationService] [node_t0] Falling back to single shard assignment since batch mode disable or multiple custom allocators set
[2024-05-21T13:54:37,547][WARN ][o.o.c.r.a.AllocationService] [node_t0] Falling back to single shard assignment since batch mode disable or multiple custom allocators set
[2024-05-21T13:54:37,626][INFO ][o.o.p.PluginsService     ] [node_t1] PluginService:onIndexModule index:[index-no-merges/WXrrQSb3QkqaLGaVAL7B0g]
[2024-05-21T13:54:40,706][WARN ][o.o.t.n.MockNioTransport ] [node_t1] Slow execution on network thread [117 milliseconds]
java.lang.RuntimeException: Slow exception on network thread
	at org.opensearch.transport.nio.MockNioTransport$TransportThreadWatchdog.maybeLogElapsedTime(MockNioTransport.java:455) [main/:?]
	at org.opensearch.transport.nio.MockNioTransport$TransportThreadWatchdog.unregister(MockNioTransport.java:446) [main/:?]
	at org.opensearch.transport.nio.TestEventHandler.handleRead(TestEventHandler.java:170) [main/:?]
	at org.opensearch.nio.NioSelector.handleRead(NioSelector.java:438) [main/:?]
	at org.opensearch.nio.NioSelector.processKey(NioSelector.java:264) [main/:?]
	at org.opensearch.nio.NioSelector.singleLoop(NioSelector.java:191) [main/:?]
	at org.opensearch.nio.NioSelector.runLoop(NioSelector.java:148) [main/:?]
	at java.base/java.lang.Thread.run(Thread.java:829) [?:?]
[2024-05-21T13:54:41,499][INFO ][o.o.c.r.a.AllocationService] [node_t0] Cluster health status changed from [YELLOW] to [GREEN] (reason: [shards started [[index-no-merges][0]]]).
[2024-05-21T13:54:41,794][WARN ][o.o.c.r.a.AllocationService] [node_t0] Falling back to single shard assignment since batch mode disable or multiple custom allocators set
[2024-05-21T13:54:41,837][INFO ][o.o.s.SnapshotsService   ] [node_t0] deleting snapshots [snapshot] from repository [cparpfrypg]
[2024-05-21T13:54:43,765][INFO ][o.o.s.SnapshotsService   ] [node_t0] snapshots [snapshot/AHbje9zlR5iEhZJx0K0DXg] deleted
[2024-05-21T13:54:43,790][INFO ][o.o.r.s.S3BlobStoreRepositoryTests] [testRequestStats] after test
[2024-05-21T13:54:43,792][INFO ][o.o.t.OpenSearchTestClusterRule] [testRequestStats] [S3BlobStoreRepositoryTests#testRequestStats]: cleaning up after test
[2024-05-21T13:54:43,843][WARN ][o.o.c.r.a.AllocationService] [node_t0] Falling back to single shard assignment since batch mode disable or multiple custom allocators set
[2024-05-21T13:54:43,851][INFO ][o.o.c.m.MetadataDeleteIndexService] [node_t0] [index-no-merges/WXrrQSb3QkqaLGaVAL7B0g] deleting index
[2024-05-21T13:54:43,851][WARN ][o.o.c.r.a.AllocationService] [node_t0] Falling back to single shard assignment since batch mode disable or multiple custom allocators set
[2024-05-21T13:54:43,906][INFO ][o.o.c.m.MetadataIndexTemplateService] [node_t0] removing template [random_index_template]
[2024-05-21T13:54:43,925][INFO ][o.o.r.RepositoriesService] [node_t0] delete repository [cparpfrypg]
[2024-05-21T13:54:43,942][INFO ][o.o.n.Node               ] [testRequestStats] stopping ...
[2024-05-21T13:54:43,948][INFO ][o.o.c.c.FollowersChecker ] [testRequestStats] FollowerChecker{discoveryNode={node_t2}{b_-9xXmqQTy5VW6fvCGllw}{GXCiImH3SHmPMtRwT8SmiQ}{127.0.0.1}{127.0.0.1:56375}{shard_indexing_pressure_enabled=true}, failureCountSinceLastSuccess=0, [cluster.fault_detection.follower_check.retry_count]=3} disconnected
[2024-05-21T13:54:43,948][INFO ][o.o.c.c.FollowersChecker ] [testRequestStats] FollowerChecker{discoveryNode={node_t1}{zEe9xRGYRoiU7kpgA-aeNQ}{3kVkAprNRqSyxi0UH30Oag}{127.0.0.1}{127.0.0.1:56373}{dimr}{shard_indexing_pressure_enabled=true}, failureCountSinceLastSuccess=0, [cluster.fault_detection.follower_check.retry_count]=3} disconnected
[2024-05-21T13:54:43,949][INFO ][o.o.c.c.FollowersChecker ] [node_t0] FollowerChecker{discoveryNode={node_t2}{b_-9xXmqQTy5VW6fvCGllw}{GXCiImH3SHmPMtRwT8SmiQ}{127.0.0.1}{127.0.0.1:56375}{shard_indexing_pressure_enabled=true}, failureCountSinceLastSuccess=0, [cluster.fault_detection.follower_check.retry_count]=3} marking node as faulty
[2024-05-21T13:54:43,951][INFO ][o.o.c.c.FollowersChecker ] [node_t0] FollowerChecker{discoveryNode={node_t1}{zEe9xRGYRoiU7kpgA-aeNQ}{3kVkAprNRqSyxi0UH30Oag}{127.0.0.1}{127.0.0.1:56373}{dimr}{shard_indexing_pressure_enabled=true}, failureCountSinceLastSuccess=0, [cluster.fault_detection.follower_check.retry_count]=3} marking node as faulty
[2024-05-21T13:54:43,956][INFO ][o.o.c.c.Coordinator      ] [node_t2] cluster-manager node [{node_t0}{tGPNeinSQ5WZPUeJ68LKpw}{dKK-5nlsQmyAqaC35b_Etg}{127.0.0.1}{127.0.0.1:56374}{dimr}{shard_indexing_pressure_enabled=true}] failed, restarting discovery
org.opensearch.transport.NodeDisconnectedException: [node_t0][127.0.0.1:56374][disconnected] disconnected
[2024-05-21T13:54:43,956][INFO ][o.o.c.c.Coordinator      ] [node_t1] cluster-manager node [{node_t0}{tGPNeinSQ5WZPUeJ68LKpw}{dKK-5nlsQmyAqaC35b_Etg}{127.0.0.1}{127.0.0.1:56374}{dimr}{shard_indexing_pressure_enabled=true}] failed, restarting discovery
org.opensearch.transport.NodeDisconnectedException: [node_t0][127.0.0.1:56374][disconnected] disconnected
[2024-05-21T13:54:43,960][INFO ][o.o.n.Node               ] [testRequestStats] stopped
[2024-05-21T13:54:43,960][INFO ][o.o.n.Node               ] [testRequestStats] closing ...
[2024-05-21T13:54:43,963][INFO ][o.o.c.s.ClusterApplierService] [node_t1] cluster-manager node changed {previous [{node_t0}{tGPNeinSQ5WZPUeJ68LKpw}{dKK-5nlsQmyAqaC35b_Etg}{127.0.0.1}{127.0.0.1:56374}{dimr}{shard_indexing_pressure_enabled=true}], current []}, term: 1, version: 27, reason: becoming candidate: onLeaderFailure
[2024-05-21T13:54:43,963][INFO ][o.o.c.s.ClusterApplierService] [node_t2] cluster-manager node changed {previous [{node_t0}{tGPNeinSQ5WZPUeJ68LKpw}{dKK-5nlsQmyAqaC35b_Etg}{127.0.0.1}{127.0.0.1:56374}{dimr}{shard_indexing_pressure_enabled=true}], current []}, term: 1, version: 27, reason: becoming candidate: onLeaderFailure
[2024-05-21T13:54:43,980][INFO ][o.o.n.Node               ] [testRequestStats] closed
[2024-05-21T13:54:43,982][INFO ][o.o.n.Node               ] [testRequestStats] stopping ...
[2024-05-21T13:54:43,994][WARN ][o.o.c.NodeConnectionsService] [node_t1] failed to connect to {node_t0}{tGPNeinSQ5WZPUeJ68LKpw}{dKK-5nlsQmyAqaC35b_Etg}{127.0.0.1}{127.0.0.1:56374}{dimr}{shard_indexing_pressure_enabled=true} (tried [1] times)
org.opensearch.transport.ConnectTransportException: [node_t0][127.0.0.1:56374] connect_exception
	at org.opensearch.transport.TcpTransport$ChannelsConnectedListener.onFailure(TcpTransport.java:1098) ~[main/:?]
	at org.opensearch.core.action.ActionListener.lambda$toBiConsumer$2(ActionListener.java:217) ~[main/:?]
	at org.opensearch.common.concurrent.CompletableContext.lambda$addListener$0(CompletableContext.java:57) ~[main/:?]
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:859) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:837) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2094) ~[?:?]
	at org.opensearch.common.concurrent.CompletableContext.completeExceptionally(CompletableContext.java:72) ~[main/:?]
	at org.opensearch.nio.SocketChannelContext.connect(SocketChannelContext.java:160) ~[main/:?]
	at org.opensearch.nio.EventHandler.handleConnect(EventHandler.java:130) ~[main/:?]
	at org.opensearch.transport.nio.TestEventHandler.handleConnect(TestEventHandler.java:139) ~[main/:?]
	at org.opensearch.nio.NioSelector.attemptConnect(NioSelector.java:446) ~[main/:?]
	at org.opensearch.nio.NioSelector.registerChannel(NioSelector.java:469) ~[main/:?]
	at org.opensearch.nio.NioSelector.setUpNewChannels(NioSelector.java:458) ~[main/:?]
	at org.opensearch.nio.NioSelector.preSelect(NioSelector.java:279) ~[main/:?]
	at org.opensearch.nio.NioSelector.singleLoop(NioSelector.java:172) ~[main/:?]
	at org.opensearch.nio.NioSelector.runLoop(NioSelector.java:148) ~[main/:?]
	at java.base/java.lang.Thread.run(Thread.java:829) [?:?]
Caused by: java.net.ConnectException: Connection refused
	at java.base/sun.nio.ch.SocketChannelImpl.checkConnect(Native Method) ~[?:?]
	at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:777) ~[?:?]
	at org.opensearch.nio.SocketChannelContext.connect(SocketChannelContext.java:157) ~[main/:?]
	... 9 more
[2024-05-21T13:54:43,995][WARN ][o.o.c.NodeConnectionsService] [node_t2] failed to connect to {node_t0}{tGPNeinSQ5WZPUeJ68LKpw}{dKK-5nlsQmyAqaC35b_Etg}{127.0.0.1}{127.0.0.1:56374}{dimr}{shard_indexing_pressure_enabled=true} (tried [1] times)
org.opensearch.transport.ConnectTransportException: [node_t0][127.0.0.1:56374] connect_exception
	at org.opensearch.transport.TcpTransport$ChannelsConnectedListener.onFailure(TcpTransport.java:1098) ~[main/:?]
	at org.opensearch.core.action.ActionListener.lambda$toBiConsumer$2(ActionListener.java:217) ~[main/:?]
	at org.opensearch.common.concurrent.CompletableContext.lambda$addListener$0(CompletableContext.java:57) ~[main/:?]
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:859) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:837) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2094) ~[?:?]
	at org.opensearch.common.concurrent.CompletableContext.completeExceptionally(CompletableContext.java:72) ~[main/:?]
	at org.opensearch.nio.SocketChannelContext.connect(SocketChannelContext.java:160) ~[main/:?]
	at org.opensearch.nio.EventHandler.handleConnect(EventHandler.java:130) ~[main/:?]
	at org.opensearch.transport.nio.TestEventHandler.handleConnect(TestEventHandler.java:139) ~[main/:?]
	at org.opensearch.nio.NioSelector.attemptConnect(NioSelector.java:446) ~[main/:?]
	at org.opensearch.nio.NioSelector.registerChannel(NioSelector.java:469) ~[main/:?]
	at org.opensearch.nio.NioSelector.setUpNewChannels(NioSelector.java:458) ~[main/:?]
	at org.opensearch.nio.NioSelector.preSelect(NioSelector.java:279) ~[main/:?]
	at org.opensearch.nio.NioSelector.singleLoop(NioSelector.java:172) ~[main/:?]
	at org.opensearch.nio.NioSelector.runLoop(NioSelector.java:148) ~[main/:?]
	at java.base/java.lang.Thread.run(Thread.java:829) [?:?]
Caused by: java.net.ConnectException: Connection refused
	at java.base/sun.nio.ch.SocketChannelImpl.checkConnect(Native Method) ~[?:?]
	at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:777) ~[?:?]
	at org.opensearch.nio.SocketChannelContext.connect(SocketChannelContext.java:157) ~[main/:?]
	... 9 more
[2024-05-21T13:54:44,005][INFO ][o.o.n.Node               ] [testRequestStats] stopped
[2024-05-21T13:54:44,005][INFO ][o.o.n.Node               ] [testRequestStats] closing ...
[2024-05-21T13:54:44,009][INFO ][o.o.n.Node               ] [testRequestStats] closed
[2024-05-21T13:54:44,011][INFO ][o.o.n.Node               ] [testRequestStats] stopping ...
[2024-05-21T13:54:44,014][INFO ][o.o.n.Node               ] [testRequestStats] stopped
[2024-05-21T13:54:44,014][INFO ][o.o.n.Node               ] [testRequestStats] closing ...
[2024-05-21T13:54:44,016][INFO ][o.o.n.Node               ] [testRequestStats] closed
[2024-05-21T13:54:44,016][INFO ][o.o.t.OpenSearchTestClusterRule] [testRequestStats] [S3BlobStoreRepositoryTests#testRequestStats]: cleaned up after test
REPRODUCE WITH: ./gradlew 'null' --tests "org.opensearch.repositories.s3.S3BlobStoreRepositoryTests.testRequestStats" -Dtests.seed=CFC6A0C8EF3E9A92 -Dtests.locale=hu -Dtests.timezone=Europe/Luxembourg -Druntime.java=11

java.lang.NullPointerException
	at __randomizedtesting.SeedInfo.seed([CFC6A0C8EF3E9A92:D04214D1DAEDB1AD]:0)
	at org.opensearch.repositories.s3.S3BlobStore.extendedStats(S3BlobStore.java:243)
	at org.opensearch.repositories.blobstore.BlobStoreRepository.stats(BlobStoreRepository.java:859)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.reduce(ReferencePipeline.java:558)
	at org.opensearch.repositories.s3.S3BlobStoreRepositoryTests.testRequestStats(S3BlobStoreRepositoryTests.java:210)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:974)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:988)
	at org.opensearch.test.OpenSearchTestClusterRule$1.evaluate(OpenSearchTestClusterRule.java:369)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
	at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
	at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
	at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
	at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
	at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
	at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
	at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
	at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
	at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at java.base/java.lang.Thread.run(Thread.java:829)

--- a/plugins/repository-s3/src/internalClusterTest/java/org/opensearch/repositories/s3/S3BlobStoreRepositoryTests.java
+++ b/plugins/repository-s3/src/internalClusterTest/java/org/opensearch/repositories/s3/S3BlobStoreRepositoryTests.java
@@ -165,7 +165,6 @@ public class S3BlobStoreRepositoryTests extends OpenSearchMockAPIBasedRepository
         return builder.build();
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/opensearch-project/OpenSearch/issues/10735")
     @Override
     public void testRequestStats() throws Exception {
         final String repository = createRepository(randomName());
@@ -249,6 +248,8 @@ public class S3BlobStoreRepositoryTests extends OpenSearchMockAPIBasedRepository
             ClusterService clusterService,
             RecoverySettings recoverySettings
         ) {
+            GenericStatsMetricPublisher genericStatsMetricPublisher = new GenericStatsMetricPublisher(10000L, 10, 10000L, 10);
+
             return new S3Repository(
                 metadata,
                 registry,
@@ -263,7 +264,7 @@ public class S3BlobStoreRepositoryTests extends OpenSearchMockAPIBasedRepository
                 false,
                 null,
                 null,
-                null
+                genericStatsMetricPublisher
             ) {
 
                 @Override

